Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -612,7 +612,40 @@ final class BuildOperationBuildSystemDelegateHandler: LLBuildBuildSystemDelegate
                 // next we want to try and scoop out any errors from the output (if reasonable size, otherwise this will be very slow),
                 // so they can later be passed to the advice provider in case of failure.
                 if output.utf8.count < 1024 * 10 {
-                    let regex = try! RegEx(pattern: #".*(error:[^\n]*)\n.*"#, options: .dotMatchesLineSeparators)
+                    do {
+                        let regex = try! RegEx(pattern: #"^(.*):(\d+):(\d+):\s*(error|warning|note|remark)\S?\s*(.*)$"#, options: .anchorsMatchLines)
+                        for match in regex.matchGroups(in: output) {
+                            if let filePath = try? AbsolutePath(validating: match[0]) {
```

Comment:
should we guard for length of match? seems safer?

Reply:
Hopefully at some point Swift will have support for regular expressions where the matches can be returned as tuples so the type system knows how many there are.